compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -323,17 +323,50 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
             }
         }
+        //issue 108713 forgotten writer warning
```

Comment:
This doesn't need to link to the issue.

Also, can you pull this into a helper method? The body of `report_no_match_method_error` is already pretty very large.

Something like `-> Option<DiagnosticBuilder<....>>` that does the first half of this `if` statement.

Reply:
Got it, thanks, I added a helper function to my last commit